proxy/logging/LogAccess.cc

```diff
@@ -20,1495 +20,2711 @@
+int
+LogAccess::marshal_client_req_squid_len(char *buf)
+{
+  if (buf) {
+    int64_t val = 0;
+    if (m_client_request) {
+      val = m_client_request->length_get() + m_http_sm->client_request_body_bytes;
+    }
+    marshal_int(buf, val);
+  }
+  return INK_MIN_ALIGN;
+}
 
 /*-------------------------------------------------------------------------
-  This case is special because it really stores 2 ints.
   -------------------------------------------------------------------------*/
 
 int
-LogAccess::marshal_cache_resp_http_version(char *buf)
+LogAccess::marshal_client_req_tcp_reused(char *buf)
 {
   if (buf) {
-    int64_t major = 0;
-    int64_t minor = 0;
-    marshal_int(buf, major);
-    marshal_int((buf + INK_MIN_ALIGN), minor);
+    int64_t tcp_reused;
+    tcp_reused = m_http_sm->client_tcp_reused;
+    marshal_int(buf, tcp_reused);
   }
-  return (2 * INK_MIN_ALIGN);
+  return INK_MIN_ALIGN;
 }
 
-LOG_ACCESS_DEFAULT_FIELD(marshal_cache_write_code, DEFAULT_INT_FIELD)
-
-LOG_ACCESS_DEFAULT_FIELD(marshal_cache_write_transform_code, DEFAULT_INT_FIELD)
-
 /*-------------------------------------------------------------------------
   -------------------------------------------------------------------------*/
 
-LOG_ACCESS_DEFAULT_FIELD(marshal_transfer_time_ms, DEFAULT_INT_FIELD)
-
-LOG_ACCESS_DEFAULT_FIELD(marshal_transfer_time_s, DEFAULT_INT_FIELD)
+int
+LogAccess::marshal_client_req_is_ssl(char *buf)
+{
+  if (buf) {
+    int64_t is_ssl;
+    is_ssl = m_http_sm->client_connection_is_ssl;
+    marshal_int(buf, is_ssl);
+  }
+  return INK_MIN_ALIGN;
+}
 
 /*-------------------------------------------------------------------------
   -------------------------------------------------------------------------*/
 
-LOG_ACCESS_DEFAULT_FIELD(marshal_file_size, DEFAULT_INT_FIELD)
-
-/*-------------------------------------------------------------------------
-  -------------------------------------------------------------------------*/
 int
-LogAccess::marshal_http_header_field(LogField::Container /* container ATS_UNUSED */, char * /* field ATS_UNUSED */, char *buf)
+LogAccess::marshal_client_req_ssl_reused(char *buf)
 {
-  DEFAULT_STR_FIELD;
+  if (buf) {
+    int64_t ssl_session_reused;
+    ssl_session_reused = m_http_sm->client_ssl_reused;
+    marshal_int(buf, ssl_session_reused);
+  }
+  return INK_MIN_ALIGN;
 }
 
 /*-------------------------------------------------------------------------
-
   -------------------------------------------------------------------------*/
+
 int
-LogAccess::marshal_http_header_field_escapify(LogField::Container /* container ATS_UNUSED */, char * /* field ATS_UNUSED */,
-                                              char *buf)
+LogAccess::marshal_client_finish_status_code(char *buf)
 {
-  DEFAULT_STR_FIELD;
+  if (buf) {
+    int code                                  = LOG_FINISH_FIN;
+    HttpTransact::AbortState_t cl_abort_state = m_http_sm->t_state.client_info.abort;
+    if (cl_abort_state == H
```

Comment:
Note: I have confirmed that this code was "pulled up" from the old `LogAccessHttp` with only changes to the scope to remove the `Http`. The diff of LogAccess.cc here is large, but mostly it is moving unchanged code.